onboard/src/pages/DocumentSigningPage.jsx

```diff
@@ -1,14 +1,53 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./DocumentSigningPage.css";
+import getSigner from "../api/signer.js";
 
 const DocumentSigningPage = () => {
+  const [document, setDocument] = useState([]);
+
+  useEffect(() => {
+    async function getData() {
+      let res = await getSigner();
+      let document;
+      if (res && res.payload) {
+        document = res.payload.data && !res.error ? res.payload.data : undefined;
+        if (document) {
+          setDocument(document);
+        }
+      }
+    }
+    getData();
+  }, []);
+
   return (
     <div>
       <center>
         <h1>Internal Document Signing</h1>
+        {document.status === "si" || document.status === "sd" ? "Signed Baby!" : "Not Signed. :("}
```

Comment:
If document is an array when it is set, document.status would not be available, right? So either res.payload.data is an object, or your implementation is incorrect here.

Reply:
res.payload.data is an object, and if I do not setDocument with an empty array, it returns an error.